### PR TITLE
Setup background job to send uploaded documents to DTS with proper file names and file path

### PR DESCRIPTION
**Pull request by spokenbird**
#### 🔗 Jira ticket
[CCAP-299](https://codeforamerica.atlassian.net/browse/CCAP-299?atlOrigin=eyJpIjoiZGY4OGNmNmY2MjBjNDY3Njg1N2U3NWQ5ZDQzNGE1MGUiLCJwIjoiaiJ9)

and

[CCAP-285](https://codeforamerica.atlassian.net/browse/CCAP-285?atlOrigin=eyJpIjoiYzEzYzAxYzdhYmMyNDAwYWE0NjJlYzFmOTIzMDkxMGMiLCJwIjoiaiJ9)

#### ✍️ Description
Adds a background job to send uploaded documents to the document transfer service. Checks if files were scanned for viruses before sending and retries check up to 5 times before throwing an exception if the file was never scanned. 

Happens concurrently in background processes using Spring Boot @Retryable and @Recover. Will send a Slack notification via Datadog alert if a file cannot be scanned or had a virus and thus cannot be sent to the DTS.

#### ✅ Completion tasks

- [X] Make sure the file was scanned for a virus before sending
- [X] Queue a background job to send file to the DTS if the file was successfully scanned

[CCAP-299]: https://codeforamerica.atlassian.net/browse/CCAP-299?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CCAP-285]: https://codeforamerica.atlassian.net/browse/CCAP-285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    private final String secretKey;
+    private final String accessKey;
+    private final String BUCKET_NAME;
+    private final S3Presigner s3Presigner;
+    private final S3Client s3Client;
+    private final Long presignedUrlDuration;
+    
+    public S3PresignService(@Value("${form-flow.aws.access_key}") String accessKey,
+            @Value("${form-flow.aws.secret_key}") String secretKey,
+            @Value("${form-flow.aws.s3_bucket_name}") String BUCKET_NAME,
```
Comment (cram-cfa) [inline src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java:40]: I personally think this is weird to have the parameter look like a constant and not be something like bucketName, but 🤷🏻 😂

File: src/main/java/org/ilgcc/app/utils/SubmissionUtilities.java [hunk 2]
```diff
@@ -140,10 +146,7 @@ public static Optional<LocalTimeRange> getTimeRangeInput(Map<String, Object> inp
   }
 
   public static String formatToStringFromLocalDate(Optional<LocalDate> date){
-    if(date.isPresent()){
-      return date.get().format(DateTimeFormatter.ofPattern("MM/dd/yyyy"));
-    }
-    return "";
+      return date.map(localDate -> localDate.format(DateTimeFormatter.ofPattern("MM/dd/yyyy"))).orElse("");
```
Comment (cram-cfa) [inline src/main/java/org/ilgcc/app/utils/SubmissionUtilities.java:149]: Could we also pull this formatter out into another constant like the ones up above?

File: src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            if (!isClean) {
+                throw new S3ObjectNotScannedException("The file at path: " + objectPath + " did not have tag 'scan-result' it has most likely not yet been scanned.");
+            }
+            taggingResult.tagSet().stream()
+                    .filter(tag -> tag.key().equals("scan-result"))
+                    .findFirst()
+                    .ifPresent(tag -> log.info("File at path: {} was scanned and found to be clean with 'scan-result' {}", objectPath, tag.value()));
+            return CompletableFuture.completedFuture(true);
+        } catch (S3Exception e) {
+            log.info("No object was found at path" + objectPath + " It's possible the object was moved to quarantine, or deleted.", e);
```
Comment (spokenbird) [inline src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java:99]: Make this a log.error

File: src/main/java/org/ilgcc/app/submission/actions/SendUploadedFileToDocumentTransferService.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                String fileName = FileNameUtility.getFileNameForUploadedDocument(submission, i + 1, userFiles.size());
+                CompletableFuture<Boolean> scannedAndCleanFuture = s3PresignService.isObjectScannedAndClean(userFile.getRepositoryPath());
+                int currentFileIndex = i;
+                scannedAndCleanFuture.thenAccept(scannedAndClean -> {
+                    if (scannedAndClean) {
+                        log.info("Sending file {} of {} to document transfer service for submission with ID: {}", currentFileIndex + 1, userFiles.size(), submission.getId());
+                        uploadedDocumentTransmissionJob.enqueueUploadedDocumentTransmissionJob(submission, userFile, fileName);
+                    }
+                }).exceptionally(e -> {
+                    log.error("The error is :", e);
```
Comment (spokenbird) [inline src/main/java/org/ilgcc/app/submission/actions/SendUploadedFileToDocumentTransferService.java:48]: Remove this extra log

File: src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            backoff = @Backoff(delay = 5000)
+    )
+    public CompletableFuture<Boolean> isObjectScannedAndClean(String objectPath) {
+        try {
+            GetObjectTaggingRequest getObjectTaggingRequest = GetObjectTaggingRequest.builder()
+                    .bucket(BUCKET_NAME)
+                    .key(objectPath)
+                    .build();
+            GetObjectTaggingResponse taggingResult = s3Client.getObjectTagging(getObjectTaggingRequest);
+            boolean isClean = taggingResult.tagSet().stream()
```
Comment (spokenbird) [inline src/main/java/org/ilgcc/app/file_transfer/S3PresignService.java:88]: Change these values to be String literal constants

**Pull request merged**